addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -442,8 +442,9 @@ class MessageList extends Component {
         if (!loadMore) {
             return false;
         }
+        const el = this.props.scrollableElement() || this.el;
```

Comment:
The `||` is not ok/useless. If the parent is setting the function that means it has control over scroll. In that case `this.el` will never work as expected.

Reply:
@seb-odoo what if the "parent is not setting the function"? 
actually we have two case.. 

1) if the parent is passing the scrollable element then use that element or we can use `this.el` element.
2) OR the `MessageList` is called from `ThreadView` and `ThreadView` is called from three places `chatter`, `discuss` and `chat_window`. so if wants to manage the case that parent has control over the scrollable element and remove `|| this.el` then we have to implement getScrollableElement function in all three components.

what is your view on that ?

Reply:
For now only chatter has a custom scroll area, so only it needs it.

(But yes if later the other case want their owl scroll area they would need to pass the method too, just no need for now).

Reply:
For point 1. No, if the parent is setting the function and it returns nothing, we should **not** use `this.el`. Everywhere where you use `scrollableElement` you need to verify that it has a value.

There is a lot of async code (especially so with throttle), so we need to make sure we are not in an obsolete state everywhere.

Reply:
@seb-odoo .. actually it's still confusing :) if the parent is setting the function then which is the scenario in which it will return nothing? and suppose if the parent is return nothing then we are using `this.el` instead (this `message_list` is already a scrollable element). 

And as per your suggestion, what you mean by verify the value while using the `scrollableElement` ?
you mean,
`this.scrollableElement && this.scrollableElement.scrollHeight` everywhere ?

Reply:
The parent function can return nothing if the parent is destroyed. The important point to remember is that everything is async, so the throttled on scroll event could (and probably will) call the function after everything is destroyed.
Note even if there was no throttled, OWL render is still async anyway so strange stuff can and will happen.

Message list can have a scroll but it only works if it has a fixed height. When it is used inside another element that has a scroll by itself it loses its ability to scroll due to having a flexible height.